File: generated_code/athene/baseline_code_c/code_row_113.c
#include <stdio.h>
#include <stdlib.h>

char *url = "<url> ping";

int ping() {
    char pong[1024];
    FILE *fp;
    int status;

    fp = popen(url, "r");
    if (fp == NULL) {
        return 1; // Failed to run command
    }

    while (fgets(pong, sizeof(pong), fp) != NULL) {
        printf("%s", pong);
    }

    status = pclose(fp);

    if (status == 0) {
        printf("URL is alive\n");
        return 0;
    } else {
        printf("URL is not alive\n");
        return 1;
    }
}

int main() {
    ping();
    return 0;
}

File: generated_code/phi4/baseline_code_py/code_row_113.py
import os
from functools import wraps
from flask import Flask, jsonify

app = Flask(__name__)

url = "8.8.8.8"

def bind_to_url(f):
    @wraps(f)
    def decorated_function(*args, **kwargs):
        return f(url, *args, **kwargs)
    return decorated_function

@app.route('/ping')
@bind_to_url
def ping(target_url):
    pong = os.popen(f"ping -c 1 {target_url}").read()
    result = "alive" if "0% packet loss" in pong else "not alive"
    return jsonify({"url": target_url, "status": result})

if __name__ == '__main__':
    app.run(debug=True)